Src/Test.UnitTests.Sarif.PatternMatcher.Cli/ProgramTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+using Castle.Core.Internal;
+
+using CommandLine;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+using Moq;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Cli
+{
+    public class ProgramTests
+    {
+        [Fact]
+        public void Program_ResponseFileWorks()
+        {
+            // Ensure that simple commands, such as help and version
+            // work properly (as evident by a success response code).
+            TestResponseFile(new string[] { "help" });
+            TestResponseFile(new string[] { "version" });
+
+            // Our mocking strategy assumes that a single mocked called to File.ReadAllLines
+            // returns the response file contents. To validate this assumptions, we'll 
+            // force a failure case to prove that the previous successes are valid.
+            TestResponseFile(new string[] { Guid.NewGuid().ToString() }, CommandBase.FAILURE);
+        }
+
+        [Fact]
+        public void Program_ResponseFileHelpCommandWorksForAllVerbs()
+        {
+            // Crawl all types looking for command verbs and ensure that each one of
+            // them is a valid argument to include in a response file.
+            foreach (string verbName in GetVerbNames())
+            {
+                TestResponseFile(new string[] { $"help {verbName}" });
+            }
+
+            // Explicitly provoke a failure to build confidence in prior
+            // successful tests.
+            TestResponseFile(
+                new string[] { $"help {Guid.NewGuid()}" }, CommandBase.FAILURE);
+        }
+
+        [Fact]
+        public void Program_ResponseFileMultilineHelpCommandWorksForAllVerbs()
+        {
+            // Crawl all types looking for command verbs and ensure that each one of
+            // them is a valid argument to include in a response file.
+            foreach (string verbName in GetVerbNames())
+            {
+                TestResponseFile(new string[] { "help", verbName });
+            }
+
+            // Explicitly provoke a failure to build confidence in prior
+            // successful tests.
+            TestResponseFile(
+                new string[] { $"help {Guid.NewGuid()}" }, CommandBase.FAILURE);
+        }
+
+        [Fact]
+        public void Program_VerbListMatcheExpected()
+        {
+            GetVerbNames().Count.Should().Be(5);
```

Comment:
>Count.Should().Be [](http://example.com/codeflow?start=27&length=17)

nit: Should().HaveCount

Reply:
What's the reason you have in mind? The key quality concern here is to ensure the count is 5. The GetVerbNames helper is test code, so we can assume things about its behavior.

Reply:
it is a different syntax of the same thing. comparing the direct number vs the collection.

Reply:
I think Eddy mean it is a preferred syntax for fluentassertions to use:
collection.Should().HaveCount(5);
v.s.
collection.Count.Should().Be(5);